### PR TITLE
feat: sync tags

### PR DESCRIPTION
**Pull request by Gibby**
Added option to support syncing tags also.

### CODE REVIEW

File: github-sync.sh [hunk 1]
```diff
@@ -29,5 +29,8 @@ git remote add tmp_upstream "$UPSTREAM_REPO"
 git fetch tmp_upstream
 git remote --verbose
 git push origin "refs/remotes/tmp_upstream/${BRANCH_MAPPING%%:*}:refs/heads/${BRANCH_MAPPING#*:}" --force
+if [[ "$SYNC_TAGS" = true ]]; then
+  git push --tags
```
Comment (core23) [inline github-sync.sh:33]: This will not work if you have changed the commit of a tag (using `git tag --force 123`)

**Review by zeke — APPROVED** [review]
This change looks good to me. @wei what do you think?

**Comment by wei** [comment]
@core23 @Gibby @zeke We are already `--force` pushing branches, should we just force push tags too (`SYNC_TAGS` is opt-in only, so I don't see too big of a problem)?

**Comment by zeke** [comment]
> should we just force push tags too 

👍🏼

**Comment by wei** [comment]
Great! I'll perform some testing and get it into a release! 😄

**Comment by wei** [comment]
@all-contributors please add @Gibby for code

**Comment by wei** [comment]
@Gibby @zeke It's been released and accessible via `v2.3.0` and `v2`. Please try it out and lmk if you see any issues 🚀

**Pull request merged**